.github/workflows/main.yml

```diff
@@ -1,13 +1,43 @@
 # Run test file with command:
 #   act pull_request -e testdata/act/pull-request.json
 
-name: Test find flags
+name: Test and Generate Docs
+
 on: pull_request
 
 jobs:
-  find-feature-flags:
+  Run_Go_Tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install pre-commit
+        run: |
+          python -m pip install --upgrade pip
+          pip install pre-commit
+
+      - name: Run pre-commit hooks
+        run: pre-commit run --all-files
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.22
+
+      - name: Install dependencies
+        run: go mod tidy
+
+      - name: Run tests
+        run: go test ./... -coverprofile=coverage.out
```

Comment:
Need to report these to datadog, you'll need to use [gotestsum to generate this information](https://github.com/launchdarkly/gonfalon/blob/main/.github/actions/run-go-tests/action.yml#L96-L102) and then [publish the generated junit](https://github.com/launchdarkly/gonfalon/blob/main/.github/actions/publish-junit/action.yml)

Reply:
I think you need to add it in TF. See [here for gonfalon example](https://github.com/launchdarkly/terraform/blob/5a0306a90a1cdb4b18e7a0d26c8e27c7b71b712b/accounts/github/launchdarkly/common/repo_gonfalon.tf#L15-L16), and it's [read here](https://github.com/launchdarkly/terraform/blob/5a0306a90a1cdb4b18e7a0d26c8e27c7b71b712b/accounts/github/launchdarkly/common/repo_gonfalon.tf#L85-L88). Maybe it's okay to re-use but also, maybe better to [like so](https://github.com/launchdarkly/terraform/blob/5a0306a90a1cdb4b18e7a0d26c8e27c7b71b712b/accounts/launchdarkly/shared-services/common/services/datadog/keys.tf#L21-L21)

Reply:
https://github.com/launchdarkly/terraform/pull/16362